modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/PinUtils.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+export const addPin = (cPins, setCpins, addNewPinState) => {
+	setCpins(
```

Comment:
I think if you move to a redux-like styles and prevent using setters as mentioned in https://github.com/liferay-frontend/liferay-portal/pull/1242/files#r669765413, it will look more easier to understand.